src/main/java/com/sap/cloudfoundry/client/facade/adapters/ServiceInstanceTypeDeserializer.java

```diff
@@ -0,0 +1,38 @@
+package com.sap.cloudfoundry.client.facade.adapters;
+
+import java.io.IOException;
+
+import org.cloudfoundry.client.v3.serviceinstances.ServiceInstanceType;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+/*
+ * TODO: This custom deserializer will be needed during switch applying of new version, after that this logic can be removed.
```

Comment:
This sentence is a bit confusing. Maybe reword it to
```
This custom deserializer is needed during rollout of the upcoming cf-java-client version 2.12.0.
It can be deleted with version 2.13.0
```